assembly/src/parsers/serde/mod.rs

```diff
@@ -0,0 +1,277 @@
+use super::{
+    ByteReader, ByteWriter, Deserializable, Instruction, Node, ProcedureId, Serializable,
+    SerializationError,
+};
+use crate::MAX_PUSH_INPUTS;
+use num_enum::TryFromPrimitive;
+
+mod deserialization;
+mod serialization;
+
+const IF_ELSE_OPCODE: u8 = 253;
```

Comment:
Not introduced in this PR: It's not guaranteed by `OpCode` that we will never overlap with these constants. Maybe add a test that fails if we successfully parse them?

Example:

```rust
#[test]
fn reserved_opcodes_wont_parse() {
  assert!(Opcode::try_from(IF_ELSE_OPCODE).is_err());
  assert!(Opcode::try_from(REPEAT_OPCODE).is_err());
  assert!(Opcode::try_from(WHILE_OPCODE).is_err());
}
```

Also, it might be worthy to move these constants to be associated with the type:

```rust
impl OpCode {
  const IF_ELSE_OPCODE: u8 = 253;
  const REPEAT_OPCODE: u8 = 254;
  const WHILE_OPCODE: u8 = 255;
}
```

Since they are implementation details, it's ok to keep them private.

Reply:
Good point! I've actually refactored to remove this constants entirely. Let me know what you think.